packages/platform/src/HttpApiClient.ts

```diff
@@ -187,11 +209,134 @@ export const make = <Groups extends HttpApiGroup.Any, ApiError, ApiR>(
             Effect.catchIf(ParseResult.isParseError, Effect.die),
             Effect.mapInputContext((input) => Context.merge(context, input))
           )
+
+        options?.onEndpoint({
+          ...onEndpointOptions,
+          endpointFn
+        })
       }
     })
-    return client as any
   })
 
+/**
+ * @since 1.0.0
+ * @category constructors
+ */
+export const make = <Groups extends HttpApiGroup.Any, ApiError, ApiR>(
+  api: HttpApi.HttpApi<Groups, ApiError, ApiR>,
+  options?: {
+    readonly transformClient?: ((client: HttpClient.HttpClient) => HttpClient.HttpClient) | undefined
+    readonly transformResponse?:
+      | ((effect: Effect.Effect<unknown, unknown, Scope>) => Effect.Effect<unknown, unknown, Scope>)
+      | undefined
+    readonly baseUrl?: string | undefined
+  }
+): Effect.Effect<
+  Simplify<Client<Groups, ApiError>>,
+  never,
+  HttpApiMiddleware.HttpApiMiddleware.Without<ApiR | HttpApiGroup.ClientContext<Groups>> | HttpClient.HttpClient
+> => {
+  const client: Record<string, Record<string, any>> = {}
+  return makeClient(api, {
+    ...options,
+    onGroup({ group }) {
+      if (group.topLevel) return
+      client[group.identifier] = {}
+    },
+    onEndpoint({ endpoint, endpointFn, group }) {
+      ;(group.topLevel ? client : client[group.identifier])[endpoint.name] = endpointFn
```

Comment:
It might make more sense to allow a predicate to be passed in.

Reply:
I didn't understand - could you explain or suggest changes)
now this callback (`makeClient.onEndpoint`) serves to perform quite diverse tasks and I'm not sure that you can do with a predicate